### PR TITLE
Fall back to CPU for unsupported libcudf binaryops in cudf-polars

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR adds logic that should trigger CPU fallback unsupported binary ops.

### CODE REVIEW

**Comment by brandon-b-miller** [comment]
@wence- I had a general thought about this - this logic is currently very rough, but broadly speaking, I think our intention here is to fallback to CPU for any ops that we can't support through libcudf. Rather than rolling our own logic to check for all the cases that should throw, what about creating a pylibcudf wrapper for `is_supported_binaryop` (moved to public in https://github.com/rapidsai/cudf/pull/15470) and using it to query if an op is supported, then raising `NotImplementedError` if not? If this seems like a reasonable idea I can push that PR forward by cutting it down a bit and then move forward here.

**Comment by wence-** [comment]
> @wence- I had a general thought about this - this logic is currently very rough, but broadly speaking, I think our intention here is to fallback to CPU for any ops that we can't support through libcudf. Rather than rolling our own logic to check for all the cases that should throw, what about creating a pylibcudf wrapper for `is_supported_binaryop` (moved to public in #15470) and using it to query if an op is supported, then raising `NotImplementedError` if not? If this seems like a reasonable idea I can push that PR forward by cutting it down a bit and then move forward here.

Yes, let's get that over the line rather than replicating functionality.

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 1]
```diff
@@ -1392,35 +1392,14 @@ def __init__(
...
-            if (
-                left.dtype.id() in dtypes.TIMELIKE_TYPES
-                and right.dtype.id() in dtypes.TIMELIKE_TYPES
-                and dtype.id() in dtypes.FLOATING_TYPES
-            ):
-                raise NotImplementedError("No dividing durations by durations")
+            raise NotImplementedError(
+                f"Operation {op.name} not supported "
+                f"for types {left.dtype.id().name} and {right.dtype.id().name} "
+                f"with output type {self.dtype}"
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:1401]: This ones needs an .id().name as well, I think

Comment (brandon-b-miller) [reply python/cudf_polars/cudf_polars/dsl/expr.py:1401]:
```suggestion
                f"with output type {self.dtype.id().name}"
```

**Review by wence- — APPROVED** [review]
Fixed the missing coverage. Thanks Brandon!

**Pull request merged**